unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/SubSections/EventsSubSection/EventsSubSectionMenu/EventsAPIController.cs

```diff
@@ -13,74 +15,45 @@ public interface IEventsAPIController
     /// <returns></returns>
-    WebRequestAsyncOperation GetAllEvents(Action<List<EventFromAPIModel>> OnSuccess, Action<string> OnFail);
+    UniTask GetAllEvents(Action<List<EventFromAPIModel>> OnSuccess, Action<string> OnFail);
 
-    /// <summary>
-    /// Register/Unregister your user in a specific event.
-    /// </summary>
-    /// <param name="eventId">Event Id.</param>
-    /// <param name="isRegistered">True for registering.</param>
-    /// <param name="OnSuccess">It will be triggered if the operation finishes successfully.</param>
-    /// <param name="OnFail">It will be triggered if the operation fails.</param>
-    /// <returns></returns>
-    UnityWebRequestAsyncOperation RegisterAttendEvent(string eventId, bool isRegistered, Action OnSuccess, Action<string> OnFail);
+    UniTask GetDetailedEvent(Action<EventFromAPIModel> OnSuccess, string eventId);
+
+    UniTask RegisterParticipation(string eventId);
+
+    UniTask RemoveParticipation(string eventId);
 }
 
 [ExcludeFromCodeCoverage]
 public class EventsAPIController : IEventsAPIController
 {
     internal const string URL_GET_ALL_EVENTS = "https://events.decentraland.org/api/events";
-    internal const string URL_POST_MESSAGE = "https://events.decentraland.org/api/message";
-    internal const string ATTEND_EVENTS_MESSAGE_TYPE = "attend";
+    private const string URL_GET_DETAILED_EVENT = "https://events.decentraland.org/api/events/{event_id}";
+    private const string URL_PARTICIPATE_EVENT = "https://events.decentraland.org/api/events/{event_id}/attendees";
 
     internal UserProfile ownUserProfile => UserProfile.GetOwnUserProfile();
 
-    public WebRequestAsyncOperation GetAllEvents(Action<List<EventFromAPIModel>> OnSuccess, Action<string> OnFail)
+    public async UniTask GetAllEvents(Action<List<EventFromAPIModel>> OnSuccess, Action<string> OnFail)
     {
-        return (WebRequestAsyncOperation) DCL.Environment.i.platform.webRequest.Get(
-            URL_GET_ALL_EVENTS,
-            OnSuccess: (webRequestResult) =>
-            {
-                EventListFromAPIModel upcomingEventsResult = JsonUtility.FromJson<EventListFromAPIModel>(webRequestResult.webRequest.downloadHandler.text);
-                OnSuccess?.Invoke(upcomingEventsResult.data);
-            },
-            OnFail: (webRequestResult) =>
-            {
-                OnFail?.Invoke(webRequestResult.webRequest.error);
-            });
+        UnityWebRequest result = await DCL.Environment.i.platform.webRequest.GetAsync(URL_GET_ALL_EVENTS, isSigned: true);
+        EventListFromAPIModel eventListFromAPIModel = Utils.SafeFromJson<EventListFromAPIModel>(result.downloadHandler.text);
+        OnSuccess?.Invoke(eventListFromAPIModel.data);
     }
 
-    public UnityWebRequestAsyncOperation RegisterAttendEvent(string eventId, bool isRegistered, Action OnSuccess, Action<string> OnFail)
+    public async UniTask GetDetailedEvent(Action<EventFromAPIModel> OnSuccess, string eventId)
```

Comment:
Do we ever need it, I don't see any references? If we want to preserve it, let's at least remove a callback